stdlib/Printf/src/Printf.jl

```diff
@@ -762,6 +849,48 @@ const UNROLL_UPTO = 16
     return pos
 end
 
+
+@inline function findonesplace(buf, lbound, rbound)
+    decimalpoint = findlast(==(UInt8('.')), buf[lbound:rbound])
+    # find last digit of rounded float
+    if isnothing(decimalpoint)
+        for i in lbound:(rbound - 1)
+            if 0x30 <= buf[i] <= 0x39 && buf[i + 1] == UInt(' ')
+                decimalpoint = i + 1
+                break
+            end
+        end
+    end
+    onesplace = isnothing(decimalpoint) ? rbound : decimalpoint - 1
+    return onesplace
+end
+
+
+function insertsep(buf, headpos, numsep, onesplace; numskip=0)
+    intlength = (onesplace - headpos + 1) - numsep - numskip
+
+    headdivlength = mod1(intlength, 3) + numskip
+    seconddiv = headpos + numsep + headdivlength
+    separation = headpos + headdivlength
+    buf[headpos:(separation - 1)] = buf[(headpos + numsep):(headpos + numsep + headdivlength - 1)]
+
+    for i in 1:numsep
+        div = seconddiv + 3 * (i - 1)
+        buf[separation] = UInt8(',')
+        buf[(separation + 1):(separation + 3)] = buf[div:(div + 2)]
+        separation += 4
+    end
+end
+
+
+countthousandsep(::Spec, x) = 0
+@inline function countthousandsep(f::Spec{T}, x) where {T <: Union{Ints, Floats}}
+    (isnan(x) || isinf(x) || base(T) != 10) && return 0
+    x2 = trunc(BigInt, round(x, digits=f.precision))
```

Comment:
I don't think this approach is going to work; creating a `BigInt` per float that we're trying to format is going to be too expensive. I think some simple benchmarks would show this to be much slower than previously.

Reply:
Would something like this work?
```
function countthousandsep(f::Spec{T}, x) where {T <: Union{Ints, Floats}}
    (isnan(x) || isinf(x) || base(T) != 10) && return 0
    x < 0 && return countthousandsep(f, -x)
    x2 = round(x, digits=f.precision)
    x2 < 1000 && return 0
    numdig = log10(x2)
    numsep = Int(div(numdig, 3))
    return numsep
end
```

Reply:
Thanks for your suggestion, @GregPlowman!
It works well except when `x` is a minimum value of a signed integer like `typemin(Int8)` because `-x` causes StackOverflow.

I've fixed the function as below, adding `x::BigInt` ver. They passed the test in my env.
```
@inline function countthousandsep(f::Spec{T}, x::BigInt) where {T <: Union{Ints, Floats}}
    (isnan(x) || isinf(x) || base(T) != 10) && return 0
    x < 0 && return countthousandsep(f, -x)
    x2 = round(x, digits=f.precision)
    x2 < 1000 && return 0
    numdig = log10(x2)
    numsep = Int(div(numdig, 3))
    return numsep
end

@inline function countthousandsep(f::Spec{T}, x) where {T <: Union{Ints, Floats}}
    (isnan(x) || isinf(x) || base(T) != 10) && return 0
    x isa Signed && x == typemin(typeof(x)) && return countthousandsep(f, - x - 1)
    x < 0 && return countthousandsep(f, -x)
    x2 = round(x, digits=f.precision)
    x2 < 1000 && return 0
    numdig = log10(x2)
    numsep = Int(div(numdig, 3))
    return numsep
end
```

I think my code is a bit lengthy. I'm open to any suggestions on how to improve it.

Reply:
@ki-chi, could you try this, also removing `x::BigInt` method. Does it fix the overflow for signed integers?
```
function countthousandsep(f::Spec{T}, x) where {T <: Union{Ints, Floats}}
    (isnan(x) || isinf(x) || base(T) != 10) && return 0
    x isa Base.BitSigned && x == typemin(x) && return countthousandsep(f, unsigned(x))
    x < 0 && return countthousandsep(f, -x)
    x2 = round(x, digits=f.precision)
    x2 < 1000 && return 0
    numdig = log10(x2)
    numsep = Int(div(numdig, 3))
    return numsep
end
```

Reply:
@GregPlowman  Yes, it has passed the test without the `x::BigInt` method. Thanks!